### PR TITLE
Allow filtering builds by commit.

### LINKED ISSUE #872: Provide an API to query the build status by commit
In order to do a check before release that everything is ok, I would like to have a way to obtain the current build status for a given commit. So, in addition to:

```
GET /api/v1/build/{id}/
```

also have this:

```
GET /api/v1/commit/{sha1}/
```

or 

```
GET /api/v1/{user}/{project}/commit/{sha1}/
```

Is this possible right now?

### PR DESCRIPTION
**Pull request by ericholscher**
Extends #3544
Fixes #872

### CODE REVIEW

File: readthedocs/restapi/views/model_views.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ class BuildViewSetBase(UserSelectViewSet):
     serializer_class = BuildSerializer
     admin_serializer_class = BuildAdminSerializer
     model = Build
-    filter_fields = ('project__slug',)
+    filter_fields = ('project__slug', 'commit')
```
Comment (humitos) [inline readthedocs/restapi/views/model_views.py]: The commit has is not unique, so by querying only by commit you could receive 2 builds from different projects. I suppose that if the user cares about that, will also apply the filter `project__slug`.

I'm assuming that we are fine.

**Comment by humitos** [comment]
Ups! I approved it but I forgot to mention that we should add this to our docs: https://docs.readthedocs.io/en/latest/api/v2.html#builds

**Comment by humitos** [comment]
There is a section there for "Query Parameters"

**Comment by ericholscher** [comment]
Good catch. Will add it. 👍

**Pull request merged**